### PR TITLE
feat: add `Window::is_decorated`

### PR DESCRIPTION
**Pull request by amrbashir**
- [X] Tested on all platforms changed
- [X] Added an entry to `CHANGELOG.md` if knowledge of this change could be valuable to users
- [X] Updated documentation to reflect any user-facing changes, including notes of platform-specific behavior
- [X] Created or updated an example program if it would help users understand this functionality
- [X] Updated [feature matrix](https://github.com/rust-windowing/winit/blob/master/FEATURES.md), if new features were added or implemented

see #2169 for context

### CODE REVIEW

File: src/window.rs [hunk 1]
```diff
@@ -701,6 +701,17 @@ impl Window {
         self.window.is_maximized()
     }
 
+    /// Gets the window's current decorations state.
+    ///
```
Comment (madsmtm) [inline src/window.rs:705]: Could you document use-cases, and relation to `set_decorations`?

**Review by madsmtm — CHANGES_REQUESTED** [review]
Nice, thanks!

File: src/window.rs (hunk 1 above)
Comment (amrbashir) [reply src/window.rs:705]: Why is documenting a use-case necessary? it is simply a getter of a state, and again I just followed the convention of `is_maximized`.

Comment (madsmtm) [reply src/window.rs]: Moving them close together is enough for me 👍

**Review by madsmtm — APPROVED** [review]
Thanks!

**Review by kchibisov — APPROVED** [review]

**Pull request merged**